[Bug] SITL doesn't build due to LibOpticalFlow permission error

### Describe the bug

SITL not building from main in the last week on WSL2. 

Maybe related to https://github.com/PX4/PX4-Autopilot/pull/25748 ?

This happens on two computers, so probably not specific to me. Relevant it of log.

```
[7/7] Linking CXX shared library libOpticalFlow.so
[26/1088] Performing install step for 'OpticalFlow'
FAILED: OpticalFlow/src/OpticalFlow-stamp/OpticalFlow-install /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-stamp/OpticalFlow-install
cd /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-build && /usr/bin/cmake --build . --target install && /usr/bin/cmake -E touch /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-stamp/OpticalFlow-install
[0/1] Install the project...
-- Install configuration: ""
-- Installing: /usr/local/lib/libOpticalFlow.so
CMake Error at cmake_install.cmake:52 (file):
  file INSTALL cannot copy file
  "/home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-build/libOpticalFlow.so"
  to "/usr/local/lib/libOpticalFlow.so": Permission denied.


FAILED: CMakeFiles/install.util
cd /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-build && /usr/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed.
[35/1088] Building CXX object src/modules/simulation...latformController.dir/MovingPlatformController.cpp.o
ninja: build stopped: subcommand failed.
```

<details>
 make px4_sitl gz_x500
[0/1088] git submodule src/drivers/gps/devices
[1/1088] git submodule src/modules/mavlink/mavlink
[2/1088] Performing configure step for 'OpticalFlow'
CMake Deprecation Warning at external/klt_feature_tracker/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Building klt_feature_tracker without catkin
-- Building OpticalFlow with OpenCV
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-build
[8/1088] git submodule src/modules/uxrce_dds_client/Micro-XRCE-DDS-Client
[25/1088] Performing build step for 'OpticalFlow'
[1/7] Building CXX object CMakeFiles/OpticalFlow.dir/src/px4flow.cpp.o
[2/7] Building CXX object CMakeFiles/OpticalFlow.dir/src/optical_flow.cpp.o
[3/7] Building CXX object CMakeFiles/OpticalFlow.dir/src/flow_px4.cpp.o
[4/7] Building CXX object klt_feature_tracker/CMakeFiles/klt_feature_tracker.dir/src/trackFeatures.cpp.o
[5/7] Building CXX object CMakeFiles/OpticalFlow.dir/src/flow_opencv.cpp.o
[6/7] Linking CXX static library klt_feature_tracker/libklt_feature_tracker.a
[7/7] Linking CXX shared library libOpticalFlow.so
[26/1088] Performing install step for 'OpticalFlow'
FAILED: OpticalFlow/src/OpticalFlow-stamp/OpticalFlow-install /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-stamp/OpticalFlow-install
cd /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-build && /usr/bin/cmake --build . --target install && /usr/bin/cmake -E touch /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-stamp/OpticalFlow-install
[0/1] Install the project...
-- Install configuration: ""
-- Installing: /usr/local/lib/libOpticalFlow.so
CMake Error at cmake_install.cmake:52 (file):
  file INSTALL cannot copy file
  "/home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-build/libOpticalFlow.so"
  to "/usr/local/lib/libOpticalFlow.so": Permission denied.


FAILED: CMakeFiles/install.util
cd /home/ubuntu/github/px4/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow-build && /usr/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed.
[35/1088] Building CXX object src/modules/simulation...latformController.dir/MovingPlatformController.cpp.o
ninja: build stopped: subcommand failed.
make: *** [Makefile:227: px4_sitl] Error 1
</details>

### To Reproduce

WSL2. Fetch latest main and run `make px4_sitl gz_x500`

### Expected behavior

It would be nice to be able to build SITL.

### Screenshot / Media

_No response_

### Flight Log

_No response_

### Software Version

_No response_

### Flight controller

_No response_

### Vehicle type

None

### How are the different components wired up (including port information)

_No response_

### Additional context

_No response_

--- a/src/modules/simulation/gz_plugins/optical_flow/optical_flow.cmake
+++ b/src/modules/simulation/gz_plugins/optical_flow/optical_flow.cmake
@@ -35,11 +35,7 @@ include(ExternalProject)
 find_package(OpenCV REQUIRED)
 
 if(NOT TARGET OpticalFlow)
-    if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-        set(OPTICAL_FLOW_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/OpticalFlow/install")
-    else()
-        set(OPTICAL_FLOW_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
-    endif()
+    set(OPTICAL_FLOW_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/OpticalFlow/install")
 
     ExternalProject_Add(OpticalFlow
         GIT_REPOSITORY https://github.com/PX4/PX4-OpticalFlow.git

